src/python/foglamp/core/server.py

```diff
@@ -40,15 +43,70 @@ def _make_app():
         routes.setup(app)
         return app
 
+    @staticmethod
+    def _make_core_app():
+        """Creates the Service management REST server Core a.k.s service registry
+
+        :rtype: web.Application
+        """
+        app = web.Application(middlewares=[middleware.error_middleware])
+        routes.core_setup(app)
+        return app
+
     @classmethod
     async def _start_scheduler(cls):
         """Starts the scheduler"""
         cls.scheduler = Scheduler()
         await cls.scheduler.start()
 
     @classmethod
+    def _start_storage(cls):
+        try:
+            # fix the directory for storage
+            STORAGE_DIR = r"/home/foglamp/Downloads/store/1010"
+            subprocess.call('./storage', cwd=STORAGE_DIR)
+        except Exception as ex:
+            _logger.exception(str(ex))
+
+        # TODO: add shutdown in stop
+
+    @classmethod
+    def _start_core(cls):
+        # https://aiohttp.readthedocs.io/en/stable/_modules/aiohttp/web.html#run_app
+        # web.run_app(cls._make_core_app(), host='0.0.0.0', port=8082)
+
+        # TODO: make management port dynamic?!
+        host_address = "localhost"  # fix?
+        core_mgt_port = 8082
+
+        ma = MultiApp()
+        ma.configure_app(cls._make_core_app(), host=host_address, port=core_mgt_port)
+        service_port = cls.request_available_port()
```

Comment:
request_available_port should not be used (because a port is only temporarily claimed in this way). instead, let the web server bind to port 0, it will claim a free port. then we ask which port the web server claimed.